frontend/src/__tests__/cypress/cypress/support/commands/application.ts

```diff
@@ -148,31 +152,34 @@ declare global {
   }
 }
 
-Cypress.Commands.add('visitWithLogin', (url, user = HTPASSWD_CLUSTER_ADMIN_USER) => {
+Cypress.Commands.add('visitWithLogin', (relativeUrl, credentials = HTPASSWD_CLUSTER_ADMIN_USER) => {
   if (Cypress.env('MOCK')) {
-    cy.visit(url);
+    cy.visit(relativeUrl);
   } else {
-    cy.intercept('GET', url, { log: false }).as('visitWithLogin');
-
-    cy.visit(url, { failOnStatusCode: false });
-
-    cy.wait('@visitWithLogin', { log: false }).then((interception) => {
-      if (interception.response?.statusCode === 403) {
-        cy.log('Do login');
-        // do login
+    let fullUrl: string;
+    if (relativeUrl) {
+      fullUrl = new URL(relativeUrl, Cypress.config('baseUrl') || '').href;
+    } else {
+      fullUrl = new URL(Cypress.config('baseUrl') || '').href;
+    }
+    cy.step(`Go to: ${fullUrl}`);
+    cy.intercept('GET', fullUrl, { log: false }).as('page');
+    cy.visit(fullUrl, { failOnStatusCode: false });
+    cy.wait('@page', { log: false }).then((interception) => {
+      const statusCode = interception.response?.statusCode;
+      if (statusCode === 403) {
```

Comment:
Could we also print the error code if there is a general error?
 
 cy.log(`Unexpected status code ${statusCode}. Throwing error.`);
throw new Error(`Failed to visit '${fullUrl}'. Status code: ${statusCode || 'unknown'}`);

Reply:
Why to print both log and error messages about the same status code ?
The error message already mentions it, so I don't think log is required.

Reply:
Those were examples, I think either or is fine really. The key point I was highlighting was that if this fails we need to know why, what the error code was and what the constructed URL is/was, otherwise it will be difficult to debug.

Reply:
Yes, good point. In line 182 it prints it via error message.